packages/orama/src/components/facets.ts

```diff
@@ -56,23 +57,64 @@ export async function getFacets(
     for (const facet of facetKeys) {
       const facetValue = facet.includes('.')
         ? (await getNested<string>(doc!, facet))!
-        : (doc![facet] as number | boolean)
-
-      // Range facets based on numbers
-      if (properties[facet] === 'number') {
-        for (const range of (facetsConfig[facet] as NumberFacetDefinition).ranges) {
-          if ((facetValue as number) >= range.from && (facetValue as number) <= range.to) {
-            if (facets[facet].values[`${range.from}-${range.to}`] === undefined) {
-              facets[facet].values[`${range.from}-${range.to}`] = 1
-            } else {
-              facets[facet].values[`${range.from}-${range.to}`]++
+        : (doc![facet] as SearchableValue)
+
+      const alreadyInsertedValues = new Set()
+      switch (properties[facet]) {
+        case 'number': {
+          for (const range of (facetsConfig[facet] as NumberFacetDefinition).ranges) {
```

Comment:
I'm wondering if this part can be abstracted as a standalone function to avoid duplication when used with `number[]` type

Reply:
I was also kinda thinking that.